Include reform

adrienpacifico opened the pull request:
Modifications in france-data, plugin-agregates and survey-manager

benjello commented on openfisca_france_data/surveys.py:
```diff
@@ -33,21 +33,33 @@
-            tax_benefit_system = TaxBenefitSystem()
+            log.info("="*10, reform , "="*10)
+            if reform is None:
+                FranceDataTaxBenefitSystem = openfisca_france_data.init_country()
```
Do not introduce new names

benjello commented on openfisca_france_data/surveys.py:
```diff
@@ -33,21 +33,33 @@
+                tax_benefit_system = france_data_tax_benefit_system
+            else:
+
+                FranceDataTaxBenefitSystem = openfisca_france_data.init_country()
```
Do not introduce new names

adrienpacifico commented:
@cbenz  told me to change the name because of possible confusions between france_data and normal tax benefit system (which happened to me).

benjello commented:
Use reference vs reform or something more general

adrienpacifico commented:
The problem is not reform vs. reference but TaxBenfitSystem vs. DataTaxBenefitSystem.

benjello commented:
There is no "data" in TaxBenefitSystem. There are SurveyScenario and classical test-case Scenario.
Clarify your problem, your goal and your implementation. I will look into it ASAP.
And write a test of what we be the result you want.

adrienpacifico commented:
openfsica_france_data/model/base.py create its own version of openfisca france tax benefit system and it is a Reform object. Thus I call that the  DataTaxBenefitSystem.

My problem : I want an easy way to run a reform instead of the reference tax benefit system.
My goal : Solve my problem ?  It's done.

My implementation : Changed test_agregates.py such that I can run a reform if I want to. You seams to want a dedicated file to do that, so I changed for test_agregates_reform.py on my computer for next commit but I am not convinced since it does the same thing as test_agregates but it include an option to include a reform.


 I made some modifications to include any reform when with test_agregates, everything seams to work.


For the FranceDataTaxBenefitSystem vs. TaxBenefitSystem issue, I really don't care it was Chistophes' suggestion, even though it makes naming a bit more explicit and might prevent future users to confound original france tax benefit system  to france data tax benefit system if the second is modified (which is for now not the case, but the implementation seams that it might be planned).

No need for me to look at it asap, just need to do not diverge too much before I leave cause it can be difficult for me to sort things correctly from Marseille.

benjello commented:
Rebase and resubmit

benjello commented:
Please close #99 
Please rebase and resubmit this reform.
I can do it by hand but you will end up with a mess
